### PR TITLE
Put all local network runner logs/files under one directory

### PR DESCRIPTION
**Pull request by holisticode**
Closes https://ava-labs.atlassian.net/browse/FRAM-216

Also "standardized" `localNetwork` receiver instances to be all `ln` against with mixed `net`

### CODE REVIEW

File: local/network.go [hunk 1]
```diff
@@ -323,12 +332,15 @@ func (ln *localNetwork) addNode(nodeConfig node.Config) (node.Node, error) {
 		return nil, fmt.Errorf("repeated node name %s", nodeConfig.Name)
 	}
 
-	// [tmpDir] is where this node's config file, C-Chain config file,
+	if ln.rootDir == "" {
+		ln.log.Warn("no network root directory defined; will create this node's runtime directory in working directory")
+	}
```
Comment (danlaine) [inline local/network.go:337]: This shouldn't ever happen, right?

File: local/network.go [hunk 2]
```diff
@@ -362,6 +374,7 @@ func (ln *localNetwork) addNode(nodeConfig node.Config) (node.Node, error) {
 
 	// Use random free API port, unless given in config
 	var apiPort uint16
+	var err error
```
Comment (danlaine) [inline local/network.go:377]:
```suggestion
	var (
		// Use random free API port, unless given in config
		apiPort uint16
		err     error
	)
```

File: local/network.go [hunk 3]
```diff
@@ -69,6 +69,8 @@ type localNetwork struct {
 	nodes map[string]*localNode
 	// List of nodes that new nodes will bootstrap from.
 	bootstrapIPs, bootstrapIDs beaconList
+	// rootDir is the root directory of all node process directories
```
Comment (danlaine) [inline local/network.go:72]:
```suggestion
	// rootDir is the root directory under which we write all node
	// logs, databases, etc.
```

File: local/network.go [hunk 4]
```diff
@@ -238,6 +241,11 @@ func newNetwork(
 		}
 	}
 
+	net.rootDir, err = os.MkdirTemp("", "avalanchego-network-runner-*")
```
Comment (danlaine) [inline local/network.go:244]:
```suggestion
	net.rootDir, err = os.MkdirTemp("", "avalanche-network-runner-*")
```
to be consistent with namind

**Review by danlaine — CHANGES_REQUESTED** [review]
A few small nits by otherwise LGTM :+1:

File: local/network.go (hunk 1 above)
Comment (holisticode) [reply local/network.go:337]: Not if used in the normal flow, but just to make sure, as this is in `addNode`, and someone could be calling it via `AddNode` without the default config?

**Review by danlaine — APPROVED** [review]

**Review by danlaine — APPROVED** [review]

**Pull request merged**